packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -133,10 +133,10 @@ export default VDataIterator.extend({
       }, {})
     },
     headersWithCustomFilters (): TableHeader[] {
-      return this.computedHeaders.filter(header => header.filter)
+      return this.computedHeaders.filter(header => header.filter && (!header.hasOwnProperty('filterable') || header.filterable === true))
```

Comment:
Probably no need in `===`

```suggestion
      return this.computedHeaders.filter(header => header.filter && (!header.hasOwnProperty('filterable') || header.filterable))
```